packages/flutter/lib/src/material/tabs.dart

```diff
@@ -821,8 +822,16 @@ class TabBar extends StatefulWidget implements PreferredSizeWidget {
 
   /// The thickness of the line that appears below the selected tab.
   ///
-  /// The value of this parameter must be greater than zero and its default
-  /// value is 2.0.
+  /// The value of this parameter must be greater than zero.
+  ///
+  /// If [ThemeData.useMaterial3] is true and [TabBar] is a primary type,
+  /// the default value is 3.0. If the provided value is less than 3.0, the
```

Comment:
Why is it overridden for values < 3.0?

Reply:
This preserves the rounded indicator shape and overrides the `2.0` indicator weight. 

https://github.com/flutter/flutter/blob/4d38f94e27d4c7691d8f4eb6cc9b068db666642d/packages/flutter/lib/src/material/tabs.dart#L715
![Screenshot 2023-05-03 at 14 17 00](https://user-images.githubusercontent.com/48603081/235901456-fdaa4715-e912-4ca6-8e3a-e4e100dabaaf.png)


If the value is less than `3.0`, this indicator shape cannot be maintained. This has rounded corners, unlike the tab indicator size indicator.

![Screenshot 2023-05-03 at 14 14 00](https://user-images.githubusercontent.com/48603081/235900978-8f64248d-358d-4554-96eb-6cb0bc3b0958.png)